### PR TITLE
bump version to 7.4.0

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/15155
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-roadmap.md [hunk 1]
```diff
@@ -272,6 +272,7 @@ You might have been waiting on some items from the last version. The following l
 
 ## Recently shipped
 
+- [TiDB 7.4.0 Release Notes](https://docs.pingcap.com/tidb/v7.4/release-7.4.0)
```
Comment (qiancai) [inline tidb-roadmap.md:275]: 这里不用在这个 PR 改了吧（因为这个 PR 要 cp 到 7.4，但 7.4 没有这个 tidb-roadmap.md 文件），和 https://github.com/pingcap/docs-cn/pull/15136 里的改动重复了

**Comment by qiancai** [comment]
tiup/tiup-component-cluster-upgrade.md 里还有一个 v7.3.0 举例需要改

**Comment by qiancai** [comment]
updated in https://github.com/pingcap/docs/pull/14997/commits/04e04e4f1ce0e039759c17a64ad10a96454e0186

**Comment by qiancai** [comment]
> tiup/tiup-component-cluster-upgrade.md 里还有一个 v7.3.0 举例需要改

updated in https://github.com/pingcap/docs/pull/14997/commits/04e04e4f1ce0e039759c17a64ad10a96454e0186

**Comment by qiancai** [comment]
please bump up version for README.md as well.

File: tidb-roadmap.md (hunk 1 above)
Comment (hfxsd) [inline tidb-roadmap.md:275]:
```suggestion
```

**Review by qiancai — APPROVED** [review]

File: develop/dev-guide-sample-application-nodejs-prisma.md [hunk 2]
```diff
@@ -267,7 +267,7 @@ void main();
 If the connection is successful, the terminal will output the version of the TiDB cluster as follows:
 
 ```
-🔌 Connected to TiDB cluster! (TiDB version: 5.7.25-TiDB-v6.6.0-serverless)
+🔌 Connected to TiDB cluster! (TiDB version: 8.0.11-TiDB-v7.4.0-serverless)
```
Comment (hfxsd) [inline develop/dev-guide-sample-application-nodejs-prisma.md:270]:
```suggestion
🔌 Connected to TiDB cluster! (TiDB version: 8.0.11-TiDB-v6.6.0-serverless)
```

File: post-installation-check.md [hunk 3]
```diff
@@ -63,7 +63,7 @@ The following information indicates successful login:
 ```sql
 Welcome to the MySQL monitor.  Commands end with ; or \g.
 Your MySQL connection id is 3
-Server version: 5.7.25-TiDB-v7.3.0 TiDB Server (Apache License 2.0) Community Edition, MySQL 5.7 compatible
+Server version: 8.0.11-TiDB-v7.4.0 TiDB Server (Apache License 2.0) Community Edition, MySQL 5.7 compatible
```
Comment (hfxsd) [inline post-installation-check.md:66]:
```suggestion
Server version: 8.0.11-TiDB-v7.4.0 TiDB Server (Apache License 2.0) Community Edition, MySQL 8.0 compatible
```

File: post-installation-check.md [hunk 4]
```diff
@@ -63,7 +63,7 @@ The following information indicates successful login:
...
 Copyright (c) 2000, 2015, Oracle and/or its affiliates. All rights reserved.
```
Comment (hfxsd) [inline post-installation-check.md:67]:
```suggestion
Copyright (c) 2000, 2023, Oracle and/or its affiliates. All rights reserved.
```

File: develop/dev-guide-sample-application-nodejs-prisma.md [hunk 5]
```diff
@@ -267,7 +267,7 @@ void main();
 If the connection is successful, the terminal will output the version of the TiDB cluster as follows:
 
 ```
-🔌 Connected to TiDB cluster! (TiDB version: 5.7.25-TiDB-v6.6.0-serverless)
+🔌 Connected to TiDB cluster! (TiDB version: 8.0.11-TiDB-v6.6.0-serverless)
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-nodejs-prisma.md:270]:
```suggestion
🔌 Connected to TiDB cluster! (TiDB version: 5.7.25-TiDB-v6.6.0-serverless)
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**